src/Tribe/Views/V2/View.php

```diff
@@ -830,4 +859,34 @@ protected function filter_view_url( $canonical, string $url ) {
 
 		return $url;
 	}
-}
\ No newline at end of file
+
+	/**
+	 * {@inheritDoc}
+	 */
+	public function found_post_ids() {
+		return $this->repository->get_ids();
+	}
+
+	/**
+	 * Sets up the View template variables.
+	 *
+	 * @since TBD
+	 *
+	 * @return array An array of Template variables for the View Template.
+	 */
+	protected function setup_template_vars(  ) {
+		$template_vars = [
+			'title'       => wp_title( null, false ),
+			'events'      => $this->repository->all(),
+			'url'         => $this->get_url( true ),
+			'prev_url'    => $this->prev_url( true ),
+			'next_url'    => $this->next_url( true ),
+			'bar_keyword' => $this->context->get( 'keyword', '' ),
```

Comment:
Why not pass these two Bar stuff as an array?

Reply:
I've modified the code to do that.